x-pack/solutions/security/plugins/security_solution_serverless/server/product_features/index.ts

```diff
@@ -53,27 +48,21 @@ export const registerProductFeatures = (
   // Cases are always enabled (both for security and AI-SOC)
   configurator.cases = getCasesProductFeaturesConfigurator(enabledProductFeatureKeys);
 
-  if (productLines[ProductLine.security]) {
-    // TODO: clarify what happens with siem migrations
+  if (productLines[ProductLine.security] && !productTiers[ProductTier.searchAiLake]) {
```

Comment:
I wouldn't hide Security fully, just the timelines and notes. Because now if the config is: 
```

    { product_line: 'security', product_tier: 'search_ai_lake' },
    { product_line: 'ai_soc', product_tier: 'search_ai_lake' },
```
    
We won't have security.